test/lib/mayaUsd/render/mayaToHydra/testDagChanges.py

```diff
@@ -20,85 +20,91 @@
 import mayaUtils
 import mtohUtils
 
-class TestDagChanges(mtohUtils.MayaHydraBaseTestCase):
+from string import digits
+
+def fullPath(nodeName):
+    return cmds.ls(nodeName, l=True)[0]
+
+class DagChangesBaseTestCase(mtohUtils.MayaHydraBaseTestCase):
+
+    IMAGE_DIFF_FAIL_THRESHOLD = 0.01
+    IMAGE_DIFF_FAIL_PERCENT = 0.2
+
+    def rprimPath(self, mayaPath):
+        return '/MayaHydraViewportRenderer/rprims/Lighted' + mayaPath.replace('|', '/')
+
+class TestDagChanges(DagChangesBaseTestCase):
     _file = __file__
 
     def setUp(self):
         self.makeCubeScene()
 
         self.grp1 = cmds.createNode('transform', name='group1')
-        self.grp1Rprim = self.rprimPath(self.grp1)
-
         self.grp2 = cmds.createNode('transform', name='group2')
-        self.grp2Rprim = self.rprimPath(self.grp2)
 
-        self.imageVersion = None
-        if maya.mel.eval("defaultShaderName") != "standardSurface1":
-            self.imageVersion = 'lambertDefaultMaterial'
+        self.imageVersion = maya.mel.eval("defaultShaderName").rstrip(digits)
 
     def test_reparent_transform(self):
         cmds.parent(self.cubeTrans, self.grp1)
-        grp1ShapeRprim = self.rprimPath(self.cubeShape)
+        grp1ShapeRprim = self.rprimPath(fullPath(self.cubeShape))
         self.assertEqual(
             grp1ShapeRprim,
             self.rprimPath("|{self.grp1}|{self.cubeTrans}|{self.cubeShape}"
                            .format(self=self)))
         cmds.refresh()
-        index = self.getIndex()
-        self.assertIn(grp1ShapeRprim, index)
-        self.assertNotIn(self.cubeRprim, index)
+        self.assertNodeNameInIndex(grp1ShapeRprim)
```

Comment:
Hydra prims created from MRenderItem's won't match exactly what's in a previously-computed render index, because the numerical suffix to e.g. StandardShadedItem will change.  The important thing to assert is that the path does not change.